modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/capability/xml/XmlModelUtils.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.capability.xml;
+
+import static org.mule.runtime.module.extension.internal.capability.xml.schema.model.SchemaConstants.CURRENT_VERSION;
+import static org.mule.runtime.module.extension.internal.capability.xml.schema.model.SchemaConstants.DEFAULT_SCHEMA_LOCATION_MASK;
+import static org.mule.runtime.module.extension.internal.capability.xml.schema.model.SchemaConstants.XSD_EXTENSION;
+import org.mule.runtime.core.util.StringUtils;
+import org.mule.runtime.extension.api.annotation.capability.Xml;
+import org.mule.runtime.extension.api.introspection.ExtensionModel;
+import org.mule.runtime.extension.api.introspection.property.XmlModelProperty;
+import org.mule.runtime.module.extension.internal.util.NameUtils;
+
+import java.util.function.Supplier;
+
+/**
+ * Utils class for parsing and generation of Xml related values of an {@link ExtensionModel extension}
+ *
+ * @since 4.0
+ */
+public final class XmlModelUtils
```

Comment:
is used by XmlModelEnricher and SchemaBuilder, which are in different packages, so this has to be public